homeassistant/components/unifi/controller.py

```diff
@@ -395,15 +394,17 @@ def shutdown(self, event: Event) -> None:
 
         Used as an argument to EventBus.async_listen_once.
         """
-        self.api.stop_websocket()
+        if self.ws_task is not None:
+            self.ws_task.cancel()
 
     async def async_reset(self) -> bool:
         """Reset this controller to default state.
 
         Will cancel any scheduled setup retry and will unload
         the config entry.
         """
-        self.api.stop_websocket()
+        if self.ws_task is not None:
+            self.ws_task.cancel()
```

Comment:
Just asking it to cancel does not stop the task. So if you say reload the integration, this task may continue to run while the integration is setting up again.

You may want to wait for it to finish and suppressing the camcelled exception.

Reply:
From your comments I ran multiple different tries and cannot get a different behaviour than expected. Looked up the CancelledError and it seems to be primary relevant to catch if you want to do something extra or ignore the cancel within the task.

Clean up in async_reset also unloads all entities so should be no subscription signalling happen even if the web socket isn't instantly cancelled. From what I know at least :)